### ISSUE TITLE
Setting/restoring variables with newlines

### ISSUE THREAD

**Issue by LucHermitte** [TACC/Lmod#659]
**Describe the bug**

Impossible to update and restore environment variables that have newlines in it (in my case `PS1`).

Note: older versions of lmod (like the 7.7.24) were able up to a certain extent, but new lines were lost by `eval` ; while `$LMOD_CMD` seemed to do the right thing, `eval` (used from `module`) looses the newlines.
 
**To Reproduce**

```lua
-- my_module/last.lua
pushenv("TOTO", "42 ".. os.getenv("TOTO"))
```

```bash
# from bash
$ export TOTO='with
new
lines'

$ printf "%s\n" "${TOTO}"
with
new
lines

# With current version of lmod (observed with versions 8.24 and 8.7.27) 
$ ml my_module
La commande « new » n'a pas été trouvée, mais peut être installée avec :

# (sorry, French locale...)

# ----------------- Note
# With older versions of lmod (tested with the 7.7.24)
$ ml my_module
$ printf "%s\n" "${TOTO}" 
42 with new lines

$ ml purge
$ printf "%s\n" "${TOTO}" 
with new lines
```

**Expected behavior**

```bash
$ ml my_module
$ printf "%s\n" "${TOTO}" 
42 with
new
lines

$ ml purge
$ printf "%s\n" "${TOTO}" 
with
new
lines
```

**Some personal investigations**

With past versions, `$LMOD_CMD bash load my_module`  used to yield:

```
...
TOTO="42 with
new
line";
export TOTO;
...
```

With current versions, it yields:

```
...
TOTO=42\ 42\ with
new
lines;
export TOTO;
...
```

With current versions of Lmod I see no workaround. With previous versions, the issue could be managed by using `source <($LMOD_CMD ...)` instead of `eval`. 

Indeed, `eval` is loosing the new line characters, while `source` does not.

```bash
$ eval $(printf "TITI='%s'\n" "${TOTO}")
$ printf "%s\n" "${TITI}" 
with new line

$ source <(printf "TITI='%s'\n" "${TOTO}")
$ printf "%s\n" "${TITI}" 
with
new
line
```

And the tests with lmod 7.7

```bash
$ source <($LMOD_CMD bash load my_module)
$ source <($LMOD_CMD purge)
```

Seems to yield the right result.

Yet, 

1. I'm not sure this is correct (no side effect with lmod 7.7), no security issues...
2. Either way, I guess it'd be better to have lmod inject newline character sequences that stay compatible with `eval` given how it works nowadays.

**Desktop (please complete the following information):**
 - OS: Linux
 - Linux distribution: Centos 7, ubuntu 18
 - Lmod Version: 8.4+

**Comment by rtmclay** [comment]
Lmod switched from using quotes to using backslash to handle spaces, single and double quotes, etc.  I tried using backslash to quote the newlines but that doesn't work.  So after some testing of various ideas, it seems that surrounding newlines in double quotes works.  At least it does for me. 

Please checkout the testing branch from the lmod github repo to see if it works for you.

**Comment by rtmclay** [comment]
This issue has been fixed in Lmod 8.7.29.  Please test to see if it works for you.

**Comment by rtmclay** [comment]
O.K. to close this issue?
